### DISCUSSION TITLE
Self referencing table rules

### DISCUSSION THREAD

**Discussion by Tilican** [pocketbase/pocketbase#4469] [Q&A] [answered]
Hey :)
First of all, thank you for this exceptional tool!
I have a small issues with particular cases.

I have 3 tables:
organizations (name, user)
users (the classic table except that everyone can list)
user_organization: (user, organization, role) which is a mapping table between user and organization with a role that is either WRITE or READ.

To be able to read an organization, one must either be its creator or have their user-organization mapping in the user_organization table.

Here's the code:
I don't know if it can work that way .... Is that correct ?

```javascript
@request.auth.id = user.id ||
(
  @collection.user_organization.organization.id = id &&
  @collection.user_organization.user.id = @request.auth.id
)
```

But now I have an issue: in the case of a user who is mapped in user_organization, how can I make the user retrieve all user_organization where they are in the same organization?

Thx you :)

**Comment by ganigeorgiev** [comment, marked as answer]
When using the `@collection.*` field you have to keep in mind that it always resolve to multiple records, aka. having `@collection.user_organization.organization.id = id` will require every `user_organization` record to have a `organization` matching the id, which is not what you want in your specific case and therefore you have to prefix the operator with `?` for any/at-least-one type of match.

If you use PocketBase v0.22.0 the above can be also simplified with the new [Back-relation](https://pocketbase.io/docs/working-with-relations/#back-relations) filter support:

```js
@request.auth.id != "" &&
(
    // is the creator of the organization
    @request.auth.id = user ||
    // OR has explicit READ permission
    (
        user_organization_via_organization.user ?= @request.auth.id &&
        user_organization_via_organization.role ?= "READ"
    )
)
```

**Reply by Tilican** [reply]
Thx for your reply I don't know I can use back relation on rules 
THX @ganigeorgiev  !!
